File: Labs/Lab04/Java/LogTable.java
/* LogTable.java displays a table of logarithms.
 *
 * Input: start, stop and increment, three doubles.
 * PRE: start < stop && increment > 0.
 * Output: A table of logarithms from start to stop,
 *         with increment as the step value.
 * Begun by: Nelesen, for CS 214 at Calvin College.
 * Completed by: AJ Vrieland (ajv234)
 * Date: 02/27/2020
 ********************************************************/
import java.util.Scanner; 		// Get input

class LogTable{

  public static void main(String[] args){
    Scanner keyboard = new Scanner(System.in);
    System.out.println("To display a table of logarithms,");
    System.out.println("enter the start, stop and increment values: ");
    double start, stop, increment;
    start = keyboard.nextDouble();
    stop = keyboard.nextDouble();
    increment = keyboard.nextDouble();
    logTable(start, stop, increment);
  }

  /* function logtTable() prints to console the log table given a start stop and increment
   * Parameter: start, stop, and increment, doubles
   * Precondition: start is less than stop
   * Return: nothing
   */
  public static void logTable(double start, double stop, double increment){
    for(double count = start; count < stop; count = count + increment){
      System.out.printf("The logarithm of %.3f is %f\n", count, Math.log10(count));
    }
  }
}

File: Labs/Lab04/Ruby/logTable.rb
#! /usr/bin/ruby
# logTable.rb displays a table of logarithms
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: AJ Vrieland (ajv234)
# Date: 02/27/2020
#
# Input:  The start, stop and increment values
# Precondition: The start value is less than the stop value,
#     and the increment is greater than 0
# Output: A table of logarithms from start to stop, with increment
#     as the step value


if __FILE__ == $0
   print "Enter the start value: "
   start = gets.chomp.to_f
   print "Enter the stop value: "
   stop = gets.chomp.to_f
   print "Enter the increment value: "
   increment = gets.chomp.to_f

   while( start <= stop)
     printf "\nThe logarithm of %f is %f", start, Math.log10(start)
     start = start + increment
   end
end
